### PR TITLE
Allow env_kwargs in make_vec_env when env ID string supplied

### PR DESCRIPTION
**Pull request by ManifoldFR**
## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: docs/misc/changelog.rst [hunk 1]
```diff
@@ -17,6 +17,7 @@ New Features:
 Bug Fixes:
 ^^^^^^^^^^
 - Fix GAE computation for on-policy algorithms (off-by one for the last value) (thanks @Wovchena)
+- Make `make_vec_env` support the `env_kwargs` argument when using an env ID str (@ManifoldFR)
```
Comment (araffin) [inline docs/misc/changelog.rst:20]:
```suggestion
- Make ``make_vec_env`` support the `env_kwargs` argument when using an env ID str (@ManifoldFR)
```

**Review by araffin — CHANGES_REQUESTED** [review]
Please add a small test to check that feature, otherwise LGTM ;)

**Comment by ManifoldFR** [comment]
> Please add a small test to check that feature, otherwise LGTM ;)

Do you know standard Gym envs that have optional kwargs? I don't have one off the top of my head...

**Comment by araffin** [comment]
>Do you know standard Gym envs that have optional kwargs? I don't have one off the top of my head...

You can always look at the source code.
For instance: https://github.com/openai/gym/blob/master/gym/envs/classic_control/continuous_mountain_car.py#L32
with `goal_velocity`
or the pendulum: https://github.com/openai/gym/blob/master/gym/envs/classic_control/pendulum.py#L14

**Comment by ManifoldFR** [comment]
> > Do you know standard Gym envs that have optional kwargs? I don't have one off the top of my head...
> 
> You can always look at the source code.
> For instance: https://github.com/openai/gym/blob/master/gym/envs/classic_control/continuous_mountain_car.py#L32
> with `goal_velocity`
> or the pendulum: https://github.com/openai/gym/blob/master/gym/envs/classic_control/pendulum.py#L14

Ah thanks, I just stumbled upon that one. Writing the test and will commit soon.

File: tests/test_utils.py [hunk 2]
```diff
@@ -65,6 +66,24 @@ def test_make_atari_env(env_id, n_envs, wrapper_kwargs):
...
 
 
+def test_vec_env_kwargs(tmp_path):
+    monitor_dir = tmp_path / "test_make_vec_env/"
+    env = make_vec_env(
+        "MountainCarContinuous-v0",
+        n_envs=1,
+        monitor_dir=monitor_dir,
+        seed=0,
+        vec_env_cls=SubprocVecEnv,
```
Comment (araffin) [inline tests/test_utils.py:76]: why not the default DummyVecEnv? and why with a monitor wrapper?

Comment (ManifoldFR) [reply tests/test_utils.py:76]: Oh I merely copied over from the next test function, but it's not necessary. I'll remove lines 74 and 76.

**Review by araffin — APPROVED** [review]
LGTM, please don't forget to reformat the code and do the checks using `make commit-checks` next time ;)

**Comment by ManifoldFR** [comment]
> LGTM, please don't forget to reformat the code and do the checks using `make commit-checks` next time ;)

Duly noted, thanks!

**Pull request merged**